packages/meditrak-server/src/routes/importSurveyResponses/importSurveyResponses.js

```diff
@@ -28,193 +28,189 @@ import {
-            // A column representing a survey response
-            try {
-              if (checkIsNewSurveyResponse(columnHeader)) {
-                if (!surveyNames) {
-                  throw new Error(
-                    'When importing new survey responses, you must specify the names of the surveys they are for',
-                  );
-                }
-                const surveyName = extractTabNameFromQuery(tabName, surveyNames);
-                const survey = await transactingModels.survey.findOne({ name: surveyName });
-                const entityCode = getInfoForColumn(sheet, columnIndex, 'Entity Code');
-                const entity = await transactingModels.entity.findOne({ code: entityCode });
-                const user = await transactingModels.user.findById(req.userId);
-                // 'Date of Data' is pulled from spreadsheet, 'Date of Survey' is current time
-                const surveyDate = getDateForColumn(sheet, columnIndex);
-                const importDate = moment();
-                const newSurveyResponse = await transactingModels.surveyResponse.create({
-                  survey_id: survey.id, // All survey responses within a sheet should be for the same survey
-                  assessor_name: `${user.first_name} ${user.last_name}`,
-                  user_id: user.id,
-                  entity_id: entity.id,
-                  start_time: importDate,
-                  end_time: importDate,
-                  data_time: stripTimezoneFromDate(surveyDate),
-                  timezone: timeZone,
-                });
-                newSurveyResponseIds[columnIndex] = newSurveyResponse.id;
-              } else {
-                // Validate that every header takes id form, i.e. is an existing or deleted response
-                await hasContent(columnHeader);
-                await takesIdForm(columnHeader);
-              }
-            } catch (error) {
-              throw new ImportValidationError(
-                `Invalid survey response id ${columnHeader} causing message: ${
-                  error.message
-                } at column ${columnIndex + 1} on tab ${tabName}`,
-              );
-            }
+    const config = { timeZone, userId, surveyNames };
+    const updatePersistor = new SurveyResponseUpdatePersistor(models);
+    const workbook = xlsx.readFile(req.file.path);
+    // Go through each sheet in the workbook and process the updated survey responses
+    const entitiesBySurveyName = await getEntitiesBySurveyName(
+      models,
+      workbook.Sheets,
+      surveyNames,
+    );
+
+    const importSurveyResponsePermissionsChecker = async accessPolicy => {
+      await assertCanImportSurveyResponses(accessPolicy, models, entitiesBySurveyName);
+    };
+
+    await req.assertPermissions(
+      assertAnyPermissions([assertBESAdminAccess, importSurveyResponsePermissionsChecker]),
+    );
+
+    for (const surveySheets of Object.entries(workbook.Sheets)) {
```

Comment:
Great job refactoring the existing code, and separating its logic in self-contained parts! 🎉 

It would be great to split this 140 liner `for` loop in smaller functions at some point - perhaps an OOP style importer class would also be appropriate

Reply:
Yeah I actually started on exactly what you suggested, but threw it away as I decided it was scope creep and I needed to focus on the main priority. I definitely think it would be a worthy place for refactor!